hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -392,7 +392,7 @@ public HoodieInstant revertReplaceCommitInflightToRequested(HoodieInstant inflig
   }
 
   private void transitionState(HoodieInstant fromInstant, HoodieInstant toInstant, Option<byte[]> data) {
-    transitionState(fromInstant, toInstant, data, false);
+    transitionState(fromInstant, toInstant, data, this.metaClient.getConsistencyGuardConfig().isConsistencyCheckEnabled());
   }
```

Comment:
Can you explain why the consistency check enabled then allows the duplicate files ?

Reply:
@danny0405 Because in the HoodieWrapperFileSystem file system, only createNewFile uses consistencyguard,
If not called, it will not take effect